javascriptv3/example_code/s3/actions/get-object-legal-hold.js

```diff
@@ -1,33 +1,63 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-import { fileURLToPath } from "url";
-import { GetObjectLegalHoldCommand, S3Client } from "@aws-sdk/client-s3";
+import {
+  GetObjectLegalHoldCommand,
+  S3Client,
+  S3ServiceException,
+} from "@aws-sdk/client-s3";
 
 /**
- * @param {S3Client} client
- * @param {string} bucketName
- * @param {string} objectKey
+ * Get an object's current legal hold status.
+ * @param {{ bucketName: string, key: string }}
  */
-export const main = async (client, bucketName, objectKey) => {
-  const command = new GetObjectLegalHoldCommand({
-    Bucket: bucketName,
-    Key: objectKey,
-    // Optionally, you can provide additional parameters
-    // ExpectedBucketOwner: "ACCOUNT_ID",
-    // RequestPayer: "requester",
-    // VersionId: "OBJECT_VERSION_ID",
-  });
+export const main = async ({ bucketName, key }) => {
+  const client = new S3Client({});
 
   try {
-    const response = await client.send(command);
+    const response = await client.send(
+      new GetObjectLegalHoldCommand({
+        Bucket: bucketName,
+        Key: key,
+        // Optionally, you can provide additional parameters
+        // ExpectedBucketOwner: "<account ID that is expected to own the bucket>",
+        // VersionId: "<the specific version id of the object to check>",
+      }),
+    );
     console.log(`Legal Hold Status: ${response.LegalHold.Status}`);
-  } catch (err) {
-    console.error(err);
+  } catch (caught) {
+    if (
+      caught instanceof S3ServiceException &&
+      caught.name === "NoSuchBucket"
+    ) {
+      console.error(
+        `Error from S3 while getting legal hold status for ${key} in ${bucketName}. The bucket doesn't exist.`,
```

Comment:
I tend to organize these like this:
```suggestion
     console.error(
        `Error from S3 while getting legal hold status for ${key} in ${bucketName}.`,
    if (
      caught instanceof S3ServiceException &&
      caught.name === "NoSuchBucket"
    ) {
      console.error(
        `The bucket doesn't exist.`,
```

Reply:
Noted. I don't like the lack of context for "The bucket doesn't exist". What was I doing when I tried to hit a bucket?